bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/PullJob.java

```diff
@@ -0,0 +1,203 @@
+                    response.getStatus());
+            return;
+        }
+
+        String responseLength = response.getHeaders().get(HttpHeader.CONTENT_LENGTH);
+        if (responseLength != null) {
+            try {
+                if (Integer.parseInt(responseLength) > HTTP_MAX_CALENDAR_SIZE) {
+                    logger.warn("Calendar is too big ({} bytes), aborting request", responseLength);
+                    response.abort(new ResponseTooBigException());
+                    return;
+                }
+            } catch (NumberFormatException e) {
+                logger.debug(
+                        "While requesting calendar Content-Length was set, but is malformed. Falling back to read-loop.",
+                        e);
+            }
+        }
+
+        File tmpTargetFile;
+        FileOutputStream tmpOutStream = null;
+        try {
+            tmpTargetFile = File.createTempFile(TMP_FILE_PREFIX, null);
+            tmpOutStream = new FileOutputStream(tmpTargetFile);
+            byte[] buffer = new byte[1024];
+            int readBytesTotal = 0;
+            InputStream httpInputStream = asyncListener.getInputStream();
+            int currentReadBytes = -1;
+            while ((currentReadBytes = httpInputStream.read(buffer)) > -1) {
+                readBytesTotal += currentReadBytes;
+                if (readBytesTotal > HTTP_MAX_CALENDAR_SIZE) {
+                    logger.warn("Calendar is too big (> {} bytes). Stopping receiving calendar.",
+                            HTTP_MAX_CALENDAR_SIZE);
+                    response.abort(new ResponseTooBigException());
+                    return;
+                }
+                tmpOutStream.write(buffer, 0, currentReadBytes);
+            }
+        } catch (IOException e) {
+            logger.warn("Not able to write temporary file with downloaded iCal.");
+            return;
+        } finally {
+            if (tmpOutStream != null) {
+                try {
+                    tmpOutStream.close();
+                } catch (IOException e) {
+                    logger.trace("Failed while closing stream on cleanup.", e);
+                }
+            }
+        }
+
+        try {
+            AbstractPresentableCalendar.create(new FileInputStream(tmpTargetFile));
+        } catch (IOException | CalendarException e) {
+            logger.warn("Not able to read downloaded iCal. Validation failed or file not readable.");
+            return;
+        }
+
+        try {
+            Files.move(tmpTargetFile.toPath(), destination.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            logger.warn("Failed to replace iCal file");
+            return;
+        }
+
+        CalendarUpdateListener currentUpdateListener = listener;
+        if (currentUpdateListener != null) {
+            try {
+                currentUpdateListener.onCalendarUpdated();
+            } catch (Exception e) {
```

Comment:
Is this useful in that context? The intention about using generic `Exception` there is we want to be failure tolerant and our regular job not to be killed because of an exception in the callback. Changing this to specific exceptions would result in an unclean interface definition.

Reply:
Well it isn't really a deal breaker, but it is a better practice explicitly list the exceptions since you are also catching RuntimeExceptions which may or may not be your intention.

Reply:
It is, as we wanted to be failure tolerant here. We know that catching `Exception` will also catch unchecked exceptions like `ArrayIndexOutOfBoundsException`, this logs the error and is at least as tolerant as possible. Also catching those exceptions allows the regular called worker to stay alive (an exception that is thrown in the Runnable would kill the runnable which may be once okay but would also kill the chance to run properly in future - if the exception is unexpected and happens only "sumetimes").